aip/general/0216.md

```diff
@@ -59,7 +59,8 @@ The field referencing the `State` enum in a resource **should** behave and be
 documented as "Output only", in accordance with [AIP-203](./0203.md).
 
 APIs **should not** allow a `State` enum to be directly updated through an
-"update" method, and **should** instead use a custom state transition method.
+"update" method (or directly set through the "create" method), and **should**
+instead use custom state transition methods.
```

Comment:
Hmm. What if there are naturally different states in which to create a resource?

For example, consider a blog post - that could be created as "draft" or "public", with a "publish" method to publish a draft one. How would you expect to express that in the create call, assuming we want to be able to create posts directly as "public" to start with? (I know our resources aren't as simple as blog posts normally, but I can imagine some similar scenarios.)

Assuming you have a good answer for this - and I suspect you do - I think it would be worth adding it in here, to avoid others having to ask.

Reply:
I would assert that in the case of this example, either we should make an exception and say that you can set the state on update (it also might make sense to go from published -> draft), _or_ we should be consistent and say that the lifecycle is consistent and you have to start in draft (which is how several of our systems, such as assessments, work).

In other words, I assert that either you should take an exception to the entire paragraph (which you can do since it is "should"), and would have anyway, or else you follow the point consistently.

Thoughts?

Reply:
I don't know - I suspect that creation is something that is likely to take a while anyway, and it's often easy to go with any given state, whereas updating from one state to another involves more process (thus justifying the custom method).

I suspect we could do with more feedback on this.

Reply:
@jgeewax Any thoughts on this?

Reply:
I think my view here is that this is reasonably well covered by the use of "should" -- this is the starting rule, but if you have a good reason to break it, then it is fine.

Reply:
I'm in agreement with Luke on this one. While you may want to be able to set a specific state at time of creation (e.g., create a post with state = published), that should be a rare case and something to be avoided generally. Instead, the lifecycle should be enforced by the service (e.g., all new items are in state "draft") and should be transitioned by other means (e.g., custom methods).

Also note that this uses "should" (not "must") and therefore leaves room to break glass in cases where someone requires different behavior for a case we've not mentioned here.